diff --git a/node_modules/@strapi/plugin-users-permissions/server/bootstrap/grant-config.js b/node_modules/@strapi/plugin-users-permissions/server/bootstrap/grant-config.js
index 013e92c..4d83e5f 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/bootstrap/grant-config.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/bootstrap/grant-config.js
@@ -29,6 +29,14 @@ module.exports = (baseURL) => ({
     callback: `${baseURL}/google/callback`,
     scope: ['email'],
   },
+  yandex: {
+    enabled: false,
+    icon: 'yandex',
+    key: '',
+    secret: '',
+    callback: `${baseURL}/yandex/callback`,
+    scope: ['email'],
+  },
   github: {
     enabled: false,
     icon: 'github',
diff --git a/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js b/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js
index 855106d..618434b 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/services/providers-registry.js
@@ -97,6 +97,19 @@ const getInitialProviders = ({ purest }) => ({
         email: body.email,
       }));
   },
+  async yandex({ accessToken }) {
+    const yandex = purest({ provider: 'yandex' });
+
+    return yandex
+      .query('oauth')
+      .get('tokeninfo')
+      .qs({ accessToken })
+      .request()
+      .then(({ body }) => ({
+        username: body.email.split('@')[0],
+        email: body.email,
+      }));
+  },
   async github({ accessToken }) {
     const github = purest({
       provider: 'github',
